File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/models/Menu.kt
package com.desktopkitchen.pos.models

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class MenuCategory(
    val id: Int,
    val name: String,
    val sort_order: Int = 0,
    val active: Boolean = true
)

@JsonClass(generateAdapter = true)
data class MenuItem(
    val id: Int,
    val category_id: Int,
    val name: String,
    val price: Double,
    val description: String? = null,
    val image_url: String? = null,
    val active: Boolean = true
)

File: apps/pos/native-windows/DesktopKitchenPOS/Models/Menu.cs
using System.Text.Json.Serialization;

namespace DesktopKitchenPOS.Models;

public record MenuCategory
{
    [JsonPropertyName("id")]
    public int Id { get; init; }

    [JsonPropertyName("name")]
    public string Name { get; init; } = "";

    [JsonPropertyName("sort_order")]
    public int SortOrder { get; init; }

    [JsonPropertyName("active")]
    public bool Active { get; init; } = true;
}

public record MenuItem
{
    [JsonPropertyName("id")]
    public int Id { get; init; }

    [JsonPropertyName("category_id")]
    public int CategoryId { get; init; }

    [JsonPropertyName("name")]
    public string Name { get; init; } = "";

    [JsonPropertyName("price")]
    public double Price { get; init; }

    [JsonPropertyName("description")]
    public string? Description { get; init; }

    [JsonPropertyName("image_url")]
    public string? ImageUrl { get; init; }

    [JsonPropertyName("active")]
    public bool Active { get; init; } = true;
}
